### ISSUE TITLE
Misalignment supervision when predicting succesor state

### ISSUE THREAD

**Issue by geekyutao** [kzl/decision-transformer#15]
Hi, amazing work! 
I find that there may be is a misalignment supervision when predicting state in https://github.com/kzl/decision-transformer/blob/master/gym/decision_transformer/training/trainer.py#L69. I think it should be
        loss = self.loss_fn(
            reward_preds, action_preds, reward_preds,
            state_target[:, 1:], action_target, reward_target[:, 1:],
        )
as DT predcits next state given current state and action in https://github.com/kzl/decision-transformer/blob/master/gym/decision_transformer/models/decision_transformer.py#L96.

Thanks.

**Comment by kzl** [comment]
Thanks for pointing this out. The trainer function likely won't properly handle masking & indexing as is, and since we don't do this in the paper I won't define behavior for this situation, but I've added this small change as a pointer for those who want to work on state/reward prediction.
